jobmanager/JobQueue.h

```diff
@@ -59,7 +60,30 @@ class JobQueue
     {
         concurrency::Semaphore sema;
         addJob<StopJob>(sema, _running);
-        sema.wait();
+
+        // Wait 1ms to this semaphore to be released, if it's not
+        // released after that we will check if this a worker thread
+        // and yield it to process other jobs that jobManager may have.
+        // If it is not an worker thread then will wait forever until semaphore is released
+        if (!sema.wait(1))
+        {
+            if (WorkerThread::isWorkerThread())
+            {
+                bool waiting = true;
+                while (waiting)
```

Comment:
I prefer 
for(bool waiting = true;wating;) 
to keep the variable inside the block